webui/src/lib/api/index.js

```diff
@@ -1119,10 +1119,13 @@ class Import {
                     "type": "common_prefix",
             }],
             "commit": {
-                "message": commitMessage,
-                "metadata": commitMetadata,
-            }
+                "message": commitMessage
+            },
+        };
+        if (Object.keys(commitMetadata).length > 0) {
+            body.commit["metadata"] = commitMetadata
```

Comment:
I prefer the previous code - when commitMetadata is undefined the json stringify will not include the property.
Will require to remove the default parameter value too.